### DISCUSSION TITLE
Expander Header Height never works! And it is too ugly

### DISCUSSION THREAD

**Discussion by sachssoft** [AvaloniaUI/Avalonia#13495] [General]
The head height of the expander is too high. I really want to make it smaller. How can I adjust the height? It never works in styles and templates. 

Trying to use margin, padding, etc. in the selector fails

`<Style Selector="Expander /template/ ToggleButton#ExpanderHeader">`

**Comment by kyuranger** [comment]
You can rewrite it base on its source code in here:https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Expander.xaml

**Comment by robloo** [comment]
It seems likely that you are using Fluent theme as it has the largest height for the expander header. Keep in mind this height is carry-over from WinUI 3 and Fluent v2 and is designed for touch device usage. I would only change the default here if your app is desktop-only.

Looking at the control template you probably want to change the `MinHeight` of the control first. My guess is that makes it work for your use cases. You can change this using the lightweight styling resource `ExpanderMinHeight` or just setting the `MinHeight` directly on the control.

https://github.com/AvaloniaUI/Avalonia/blob/52cbe29916fb8ab62837257cbb8e5d63a1e4ed37/src/Avalonia.Themes.Fluent/Controls/Expander.xaml#L218

https://github.com/AvaloniaUI/Avalonia/blob/52cbe29916fb8ab62837257cbb8e5d63a1e4ed37/src/Avalonia.Themes.Fluent/Controls/Expander.xaml#L231

However, if that still isn't enough, I wouldn't retemplate the full control just yet. I would just change the header `ToggleButton` control theme.

https://github.com/AvaloniaUI/Avalonia/blob/52cbe29916fb8ab62837257cbb8e5d63a1e4ed37/src/Avalonia.Themes.Fluent/Controls/Expander.xaml#L67-L116
